### PR TITLE
Create a single header for Search

### PR DESCRIPTION
**Pull request by aprokop**
Not sure if I caught all references, but not sure of the way to detect that.

### CODE REVIEW

**Comment by sslattery** [comment]
Looks like timeouts and a number of warnings on the Cuda 9 build

File: packages/Search/src/DTK_Search.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef DTK_SEARCH_HPP
+#define DTK_SEARCH_HPP
+
+#include <DTK_DistributedSearchTree.hpp>
+#include <DTK_LinearBVH.hpp>
```
Comment (Rombur) [inline packages/Search/src/DTK_Search.hpp]: Do we want to only put these two headers or should we put all of them? I realize that the others should be included by these two but what happens if we change the headers later on? I don't have an opinion, just asking

Comment (aprokop) [reply packages/Search/src/DTK_Search.hpp]: As a rule of thumb, we could add all but `*Details*`.

**Comment by dalg24** [comment]
retest this please

**Comment by aprokop** [comment]
I added the extra non-Details headers.

**Pull request merged**